### PR TITLE
Make Security number clickable

### LINKED ISSUE #296: Make "SECURITY-x" warnings clickable
### Description of the bug

Taking https://plugin-health.jenkins.io/scores/build-metrics for example, the `security` probe reports 3 open security warnings:
Screenshot 2023-04-15 at 10 42 13
I would recommend making these clickable because there's no way to check the details on jenkins.io without additional research.
The link could forward to the same location as the security warning banner on https://plugins.jenkins.io/build-metrics/ does.

### PR DESCRIPTION
**Pull request by AayushSaini101**
### Description

Closes #296 

Make Security number clickable, it will help the maintainers and developers to know more about the security issue

```[tasklist]
- [x] If an issue exists, it is well described and linked in the description
- [x] The description of this pull request is detailed and explains why this pull request is needed
```

### CODE REVIEW

File: war/src/main/resources/templates/scores/details.html [hunk 1]
```diff
@@ -87,10 +92,26 @@ <h3>Probes results</h3>
...
+                              name="failure"></ion-icon>
+                </td>
+                <td>
+                    <span
+                        th:if="${probeResult.id == 'security' && probeResult.message != 'No known security vulnerabilities.'}">
+                        <span th:each="securityNumber : ${#strings.arraySplit(probeResult.message, ' ,')}"
+                              th:inline="text">
+                            <a th:text="${securityNumber}"
+                               th:href="'https://www.jenkins.io/security/issue/' + ${securityNumber}"  >
+                            </a>
```
Comment (alecharp) [inline war/src/main/resources/templates/scores/details.html:107]: This looks ok but some values are not the security issue id but the link to the advisory.
I'm looking for a plugin with such a result.

**Review by alecharp — CHANGES_REQUESTED** [review]
This won't work on https://plugin-health.jenkins.io/scores/aws-codecommit-trigger where ID of security issue includes `-1`

**Comment by AayushSaini101** [comment]
> This won't work on https://plugin-health.jenkins.io/scores/aws-codecommit-trigger where ID of security issue includes `-1`

@alecharp we can trim the -[n] after the security id

**Comment by AayushSaini101** [comment]
> > > This won't work on https://plugin-health.jenkins.io/scores/aws-codecommit-trigger where ID of security issue includes `-1`
> > 
> > 
> > @alecharp we can trim the -[n] after the security id
> 
> Not really because if you try the replacement manually, you'll see that the `-1` is translated into `(1)` in the end. Without this, the HTML anchor is not working and the correct security advisory is not shown directly.



> > > This won't work on https://plugin-health.jenkins.io/scores/aws-codecommit-trigger where ID of security issue includes `-1`
> > 
> > 
> > @alecharp we can trim the -[n] after the security id
> 
> Not really because if you try the replacement manually, you'll see that the `-1` is translated into `(1)` in the end. Without this, the HTML anchor is not working and the correct security advisory is not shown directly.

https://www.jenkins.io/security/issue/SECURITY-3101

**Comment by alecharp** [comment]
and the `-2`?

**Comment by AayushSaini101** [comment]
> and the `-2`?

For the simple approach we can provide the same URL:https://www.jenkins.io/security/issue/SECURITY-3101 because if we look at the page the centre page is shows the SECURITY-3101 -2 ,
image

**Comment by alecharp** [comment]
I don't really like that but I won't fight it to much because this web ui is not the main one. 
This would also need to be done, but correctly, on plugins.jenkins.io.

**Comment by AayushSaini101** [comment]
> I don't really like that but I won't fight it to much because this web ui is not the main one. This would also need to be done, but correctly, on plugins.jenkins.io.

Thanks @alecharp let me research more on this, I will try to find more about this

**Comment by AayushSaini101** [comment]
> > SECURITY-3101 -2 @alecharp
> > From the update.centre json data we can get the appropriate URL
> 
> image

I am looking into it how can use it

**Comment by alecharp** [comment]
@AayushSaini101 @aayushRedHat are you the same person? If so, please make sure you configure your Git username and email to be consistent through your contributions on this project. Also, please, try not to force push when the review process has started; this is making things harder has we have to review everything again.

File: checkstyle-10.12.4-all.jar
Comment (alecharp) [inline checkstyle-10.12.4-all.jar:1]: Please don't import binaries like those in the repository.

File: core/src/main/java/io/jenkins/pluginhealth/scoring/model/updatecenter/SecurityWarning.java [hunk 2]
```diff
@@ -29,6 +29,11 @@
 public record SecurityWarning(
     String id,
     String name,
+
+    String url,
     List<SecurityWarningVersion> versions
 ) {
+    public SecurityWarning(String name, String url) {
+        this(null, name, url, List.of());
+    }
```
Comment (alecharp) [inline core/src/main/java/io/jenkins/pluginhealth/scoring/model/updatecenter/SecurityWarning.java:38]: I don't think we want to get rid of the ID, so please don't initialize it empty.
 Also, initializing the `versions` field to an empty list will compromise the code which detects if the current version of the plugin is concerned by the security issue.

```suggestion
```

File: core/src/main/java/io/jenkins/pluginhealth/scoring/model/updatecenter/SecurityWarning.java [hunk 3]
```diff
@@ -29,6 +29,11 @@
 public record SecurityWarning(
     String id,
     String name,
+
```
Comment (alecharp) [inline core/src/main/java/io/jenkins/pluginhealth/scoring/model/updatecenter/SecurityWarning.java:32]:
```suggestion
```

File: core/src/main/java/io/jenkins/pluginhealth/scoring/probes/KnownSecurityVulnerabilityProbe.java [hunk 4]
```diff
@@ -55,7 +56,7 @@ protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
                 final Matcher matcher = pattern.matcher(plugin.getVersion().toString());
                 return matcher.find();
             }))
-            .map(SecurityWarning::id)
+            .map(SecurityWarning::url)
```
Comment (alecharp) [inline core/src/main/java/io/jenkins/pluginhealth/scoring/probes/KnownSecurityVulnerabilityProbe.java:59]: see if you can map it to a structure to store both `id` and `url`.

File: war/src/main/java/io/jenkins/pluginhealth/scoring/probes/ProbeEngine.java
Comment (alecharp) [inline war/src/main/java/io/jenkins/pluginhealth/scoring/probes/ProbeEngine.java:1]: please don't submit code that are not related to the task.

File: war/src/main/java/io/jenkins/pluginhealth/scoring/schedule/ProbeEngineScheduler.java [hunk 5]
```diff
@@ -42,9 +42,9 @@ public ProbeEngineScheduler(ProbeEngine probeEngine, ScoringEngine scoringEngine
         this.scoringEngine = scoringEngine;
     }
 
-    @Scheduled(cron = "${app.cron.probe-engine}", zone = "UTC")
+   @Scheduled(cron = "${app.cron.probe-engine}", zone = "UTC")
```
Comment (alecharp) [inline war/src/main/java/io/jenkins/pluginhealth/scoring/schedule/ProbeEngineScheduler.java:45]: please don't

File: war/src/main/java/io/jenkins/pluginhealth/scoring/schedule/ProbeEngineScheduler.java [hunk 6]
```diff
@@ -42,9 +42,9 @@ public ProbeEngineScheduler(ProbeEngine probeEngine, ScoringEngine scoringEngine
...
     public void run() throws IOException {
-        probeEngine.run();
+     //   probeEngine.run();
         scoringEngine.run();
```
Comment (alecharp) [inline war/src/main/java/io/jenkins/pluginhealth/scoring/schedule/ProbeEngineScheduler.java:48]: If you want to change the probe result stored in the database, you need the `probeEngine` to be ran.

File: war/src/main/resources/application.yaml
Comment (alecharp) [inline war/src/main/resources/application.yaml:1]: Please don't.
Set the environment variable in you IDE launcher so you don't change the file content.

**Review by alecharp — CHANGES_REQUESTED** [review]

File: checkstyle-10.12.4-all.jar
Comment (AayushSaini101) [reply checkstyle-10.12.4-all.jar:1]: @alecharp Sorry for this, I am updating

File: core/src/main/java/io/jenkins/pluginhealth/scoring/probes/KnownSecurityVulnerabilityProbe.java (hunk 4 above)
Comment (AayushSaini101) [reply core/src/main/java/io/jenkins/pluginhealth/scoring/probes/KnownSecurityVulnerabilityProbe.java:59]: @alecharp Do we need to show the front-end part like this ?

SECURITY-1511  <URL>

Comment (alecharp) [reply core/src/main/java/io/jenkins/pluginhealth/scoring/probes/KnownSecurityVulnerabilityProbe.java:59]: It would be great to be able to show the ID as an hyperlink yes.

Comment (aayushRedHat) [reply core/src/main/java/io/jenkins/pluginhealth/scoring/probes/KnownSecurityVulnerabilityProbe.java:59]: > It would be great to be able to show the ID as an hyperlink yes.

Thanks that means 
SECURITY-1511 [SECURITY-1511](https://jenkins.io/security/advisory/2020-03-09/#SECURITY-1511)

**Comment by AayushSaini101** [comment]
Closing this PR, Creating new PR with correct account
